beacon-chain/blockchain/testing/mock.go

```diff
@@ -444,3 +444,8 @@ func (s *ChainService) HeadSyncContributionProofDomain(_ context.Context, _ type
 func (s *ChainService) IsOptimistic(_ context.Context) (bool, error) {
 	return false, nil
 }
+
+// IsOptimisticForRoot mocks the same method in the chain service.
+func (s *ChainService) IsOptimisticForRoot(_ context.Context, _ [32]byte, _ types.Slot) (bool, error) {
```

Comment:
We might get deep src errors for the `_`s. We'll see. Feel free to merge if not

Reply:
The other functions above work like this.. i think it would pass...